packages/@aws-cdk/aws-events/lib/rule.ts

```diff
@@ -119,7 +88,7 @@ export class Rule extends Resource implements IRule {
   private readonly _xEnvTargetsAdded = new Set<string>();
 
   constructor(scope: Construct, id: string, props: RuleProps = { }) {
-    super(scope, id, {
+    super(determineRuleScope(scope, props), id, {
```

Comment:
Tests for this?

Reply:
This is verified by every single test that uses any `Rule`, of which there are hundreds, and also the new `"allows using a new Repository from another Stack as a source of the Pipeline, with Events"` test that was added in this PR.

Reply:
Fair. I can let that go.

Reply:
Can we get an integration test on this, though? It seems like exactly the sort of thing that we'd want integ tests for.

Reply:
Sure 🙂.